spec/abstractops.html

```diff
@@ -86,6 +86,28 @@ <h1>GetNumberOption ( _options_, _property_, _minimum_, _maximum_, _fallback_ )<
     </emu-alg>
   </emu-clause>
 
+  <emu-clause id="sec-getstringornumberoption" aoid="GetStringOrNumberOption">
+    <h1>GetStringOrNumberOption ( _options_, _property_, _stringValues_, _minimum_, _maximum_, _fallback_ )</h1>
+    <p>
+      The abstract operation GetStringOrNumberOption extracts the value of the property named _property_ from the provided _options_ object.
+      If the value is a Number value, the operation checks whether it is in the allowed range.
+      Otherwise, the operation converts it to a string and checks whether it is one of a List of allowed _stringValues_, and fills in a _fallback_ value if necessary.
+      If _stringValues_ is *undefined*, there is no fixed set of values and any is permitted.
+    </p>
+    <emu-alg>
+      1. Assert: Type(_options_) is Object.
+      1. Let _value_ be ? Get(_options_, _property_).
+      1. If _value_ is *undefined*, return _fallback_.
+      1. Set _value_ to ? ToPrimitive(_value_, ~number~).
+      1. If Type(_value_) is Number, then
+        1. If _value_ is *NaN*, or _value_ &lt; _minimum_, or _value_ &gt; _maximum_, throw a *RangeError* exception.
+        1. Return floor(ℝ(_value_)).
+      1. Set _value_ to ? ToString(_value_).
```

Comment:
Calling ToString on the result of ToPrimitive seems like a bad idea to me. If we don't have precedent on the TC39 side, I'd like to follow WebIDL precedent, which comes down to just dropping to ToPrimitive call (so use the number if Type is number, and ToString otherwise).

Reply:
There is no precedent that I'm aware of of the TC39 side, so that sounds fine to me.